### ISSUE TITLE
Instagram api key

### ISSUE THREAD

**Issue by adsgm** [iatek/jquery-socialist#51]
I'm trying to use the Instagram service with the plug-in, but I have no idea how to get my Instagram API key. Any help will be appreciated.

**Comment by iatek** [comment]
If you login to instagram.com and then go to http://instagram.com/developer/ and click 'Register your Application' it should be easy to follow from there.

**Comment by adsgm** [comment]
I did that, used the client id and client secret, and none of them worked. I hope there's a solution.

**Comment by iatek** [comment]
Is the plugin working for you otherwise? Do you get no results from Instagram or an error?

**Comment by justinjams** [comment]
@adsgm Socialist is really asking for a client id for the apiKey parameter, and the id parameter is a tag to search for. see the second call on http://instagram.com/developer/endpoints/tags/

**Comment by Jaxattack** [comment]
@justinjams Could you elaborate please? I'm wanting to display a clients instagram feed on his website, but am unsure how to generate/grab the apiKey to parse into socialist? Will I need my client to log in and register the app from his own account?

**Comment by justinjams** [comment]
@Jaxattack Yeah, your client's account should generate a client_id tied to that account. Once it's logged in, go to http://instagram.com/developer/clients/manage/ and register a new client. The client here is your website. After you've completed that, you'll see a client_id on this page.

So you'll give Socialist some configuration object with a property like:

<pre><code>networks: [
            {name:'instagram',id:'<<some relevant tag>>',apiKey:'<<client_id>>'}
        ]</code></pre>

client_id you have, but some relevant tag is necessary because Socialist makes this call:

<pre><code>https://api.instagram.com/v1/tags/|id|/media/recent?client_id=|apiKey|</code></pre>

So, really, Socialist is using the client_id only for auth, and then searching for some tag. If you just want the client's own stream, you'll need to modify Socialist.

I was able to do this without needing to make too many changes. Mainly, I changed the call on jquery.socialist.js:465 (or something) from the above snip to:

<pre><code>https://api.instagram.com/v1/users/|id|/media/recent/?access_token=|apiKey|&count=|num|</code></pre>

So now it's taking an access token -- also generated through Instagram at http://instagram.com/developer/authentication/. It's also taking a user id, which is not the same as a username. This gets returned at some point during the process I linked above, when it's returning your token. I also added a count parameter because the original call doesn't seem to obey maxResults like it should. 

Here's a gist with my modified socialist and example implementation: https://gist.github.com/justinjams/7112513

Hope this helps!
